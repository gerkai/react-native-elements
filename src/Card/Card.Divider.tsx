import React from 'react';
import { StyleSheet } from 'react-native';
import Divider, { DividerProps } from '../Divider';
import { RneFunctionComponent } from '../helpers';

export type CardDividerProps = DividerProps;

/** Add divider to the card which acts as a separator between elements.
 * This, Receives all [Divider](divider#props) props. */
export const CardDivider: RneFunctionComponent<CardDividerProps> = ({
  style,
  ...props
}) => (
  <Divider style={StyleSheet.flatten([styles.divider, style])} {...props} />
);

const styles = StyleSheet.create({
  divider: {
    marginBottom: 15,
  },
});

CardDivider.displayName = 'Card.Divider';
