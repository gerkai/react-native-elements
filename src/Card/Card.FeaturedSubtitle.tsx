import React from 'react';
import { Platform, StyleSheet, TextStyle } from 'react-native';
import normalize from '../helpers/normalizeText';
import { fonts } from '../config';
import Text, { TextProps } from '../Text';
import { RneFunctionComponent } from '../helpers';

export type CardFeaturedSubtitleProps = TextProps;

/** Add a featured subtitle to the Card.
 * This, Receives all [Text](text#props) props. */
export const CardFeaturedSubtitle: RneFunctionComponent<CardFeaturedSubtitleProps> = ({
  theme,
  style,
  ...rest
}) => (
  <Text
    style={
      StyleSheet.flatten([
        {
          fontSize: normalize(13),
          marginBottom: 8,
          color: theme?.colors?.white,
          ...Platform.select({
            android: {
              ...fonts.android.black,
            },
            default: {
              fontWeight: '400',
            },
          }),
        },
        style,
      ]) as TextStyle
    }
    {...rest}
  />
);

CardFeaturedSubtitle.displayName = 'Card.FeaturedSubtitle';
