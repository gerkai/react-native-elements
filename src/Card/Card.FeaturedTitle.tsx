import React from 'react';
import { Platform, StyleSheet, TextStyle } from 'react-native';
import normalize from '../helpers/normalizeText';
import { fonts } from '../config';
import Text, { TextProps } from '../Text';
import { RneFunctionComponent } from '../helpers';

type CardFeaturedTitleProps = TextProps;

/** Add a featured title to the Card.
 * This, Receives all [Text](text.md#props) props. */
export const CardFeaturedTitle: RneFunctionComponent<CardFeaturedTitleProps> = ({
  theme,
  style,
  ...props
}) => (
  <Text
    style={
      StyleSheet.flatten([
        {
          fontSize: normalize(18),
          marginBottom: 8,
          color: theme?.colors?.white,
          ...Platform.select({
            android: {
              ...fonts.android.black,
            },
            default: {
              fontWeight: '800',
            },
          }),
        },
        style,
      ]) as TextStyle
    }
    {...props}
  />
);

CardFeaturedTitle.displayName = 'Card.FeaturedTitle';
