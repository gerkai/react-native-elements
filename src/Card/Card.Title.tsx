import React from 'react';
import { Platform, StyleSheet, TextStyle } from 'react-native';
import normalize from '../helpers/normalizeText';
import { fonts } from '../config';
import Text, { TextProps } from '../Text';
import { RneFunctionComponent } from '../helpers';

type CardTitleProps = TextProps;

/** Add a general title to the Card.
 * This, Receives all [Text](text#props) props. */
export const CardTitle: RneFunctionComponent<CardTitleProps> = ({
  style,
  theme,
  ...props
}) => (
  <Text
    testID="cardTitle"
    style={
      StyleSheet.flatten([
        {
          fontSize: normalize(14),
          color: theme?.colors?.grey1,
          ...Platform.select({
            android: {
              ...fonts.android.black,
            },
            default: {
              fontWeight: 'bold',
            },
          }),
          textAlign: 'center',
          marginBottom: 15,
        },
        style,
      ]) as TextStyle
    }
    {...props}
  />
);

CardTitle.displayName = 'Card.Title';
