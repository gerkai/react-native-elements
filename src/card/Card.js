import React from 'react';
import PropTypes from 'prop-types';
import { View, Platform, StyleSheet } from 'react-native';

import { withTheme } from '../config';

import CardTitle from './CardTitle';
import CardDivider from './CardDivider';
import CardImage from './CardImage';
import CardFeaturedTitle from './CardFeaturedTitle';
import CardFeaturedSubtitle from './CardFeaturedSubtitle';

const Card = (props) => {
  const {
    children,
    containerStyle,
    wrapperStyle,
    theme,
    ...attributes
  } = props;

  return (
    <View
      {...attributes}
      style={StyleSheet.flatten([
        styles.container(theme),
        containerStyle && containerStyle,
      ])}
    >
      <View
        style={StyleSheet.flatten([
          styles.wrapper,
          wrapperStyle && wrapperStyle,
        ])}
      >
        {children}
      </View>
    </View>
  );
};

Card.propTypes = {
  containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  wrapperStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  overlayStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  theme: PropTypes.object,
};

const styles = {
  container: (theme) => ({
    backgroundColor: theme.colors.white,
    borderWidth: 1,
    padding: 15,
    margin: 15,
    marginBottom: 0,
    borderColor: theme.colors.grey5,
    ...Platform.select({
      android: {
        elevation: 1,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: { height: 0, width: 0 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  }),
  wrapper: {
    backgroundColor: 'transparent',
  },
};

export { Card };
const ThemedCard = withTheme(Card, 'Card');
ThemedCard.Divider = CardDivider;
ThemedCard.Image = CardImage;
ThemedCard.Title = CardTitle;
ThemedCard.FeaturedTitle = CardFeaturedTitle;
ThemedCard.FeaturedSubtitle = CardFeaturedSubtitle;
export default ThemedCard;
