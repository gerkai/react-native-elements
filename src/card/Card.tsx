import React from 'react';
import { View, Platform, StyleSheet, StyleProp, ViewStyle } from 'react-native';
import { withTheme } from '../config';
import CardTitle from './CardTitle';
import CardDivider from './CardDivider';
import CardImage from './CardImage';
import CardFeaturedTitle from './CardFeaturedTitle';
import CardFeaturedSubtitle from './CardFeaturedSubtitle';
import { Theme } from '../config/theme';

export type CardProps = {
  containerStyle?: StyleProp<ViewStyle>;
  wrapperStyle?: StyleProp<ViewStyle>;
  theme?: Theme;
};

interface Card extends React.FunctionComponent<CardProps> {}

const Card: Card = (props) => {
  const {
    children,
    containerStyle,
    wrapperStyle,
    theme,
    ...attributes
  } = props;

  return (
    <View
      {...attributes}
      style={StyleSheet.flatten([
        {
          backgroundColor: theme.colors.white,
          borderWidth: 1,
          padding: 15,
          margin: 15,
          marginBottom: 0,
          borderColor: theme.colors.grey5,
          ...Platform.select({
            android: {
              elevation: 1,
            },
            default: {
              shadowColor: 'rgba(0,0,0, .2)',
              shadowOffset: { height: 0, width: 0 },
              shadowOpacity: 1,
              shadowRadius: 1,
            },
          }),
        },
        containerStyle && containerStyle,
      ])}
    >
      <View
        style={StyleSheet.flatten([
          styles.wrapper,
          wrapperStyle && wrapperStyle,
        ])}
      >
        {children}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: 'transparent',
  },
});

export { Card };

const ThemedCard = Object.assign(withTheme(Card, 'Card'), {
  Divider: CardDivider,
  Image: CardImage,
  Title: CardTitle,
  FeaturedTitle: CardFeaturedTitle,
  FeaturedSubtitle: CardFeaturedSubtitle,
});
export default ThemedCard;
