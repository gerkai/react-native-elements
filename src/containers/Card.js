import React from 'react'
import { View, StyleSheet, Platform } from 'react-native'
import fonts from '../config/fonts'
import colors from '../config/colors'
import Text from '../text/Text'
import Divider from '../Divider'
let styles = {}

const Card = ({children, flexDirection, containerStyle, wrapperStyle, title, titleStyle, dividerStyle, fontFamily}) => (
  <View style={[styles.container, containerStyle && containerStyle]}>
    <View style={[styles.wrapper, wrapperStyle && wrapperStyle, flexDirection && {flexDirection}]}>
      {
        title && (
          <View>
            <Text style={[
              styles.cardTitle,
              titleStyle && titleStyle,
              fontFamily && {fontFamily}
            ]}>{title}</Text>
            <Divider style={[styles.divider, dividerStyle && dividerStyle]} />
          </View>
        )
      }
      {children}
    </View>
  </View>
)

styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderColor: colors.grey5,
    borderWidth: 1,
    borderRadius: 3,
    padding: 15,
    margin: 15,
    marginBottom: 0,
    ...Platform.select({
      ios: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: {height: 0, width: 0},
        shadowOpacity: 1,
        shadowRadius: 1
      },
      android: {
        elevation: 1
      }
    })
  },
  wrapper: {
    backgroundColor: 'transparent'
  },
  divider: {
    marginBottom: 15
  },
  cardTitle: {
    fontFamily: fonts.ios.bold,
    textAlign: 'center',
    marginBottom: 15
  }
})

export default Card
