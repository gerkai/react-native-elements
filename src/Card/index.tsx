import { withTheme } from '../config';
import { RneFunctionComponent } from '../helpers';
import { CardBase, CardBaseProps } from './Card';
import { CardDivider } from './Card.Divider';
import { CardFeaturedSubtitle } from './Card.FeaturedSubtitle';
import { CardFeaturedTitle } from './Card.FeaturedTitle';
import { CardImage } from './Card.Image';
import { CardTitle } from './Card.Title';

const ThemedCardDivider = withTheme(CardDivider, 'CardDivider');
const ThemedCardFeaturedSubtitle = withTheme(
  CardFeaturedSubtitle,
  'CardFeaturedSubtitle'
);
const ThemedCardFeaturedTitle = withTheme(
  CardFeaturedTitle,
  'CardFeaturedTitle'
);
const ThemedCardImage = withTheme(CardImage, 'CardImage');
const ThemedCardTitle = withTheme(CardTitle, 'CardTitle');

export type CardProps = RneFunctionComponent<CardBaseProps> & {
  Divider: typeof ThemedCardDivider;
  Image: typeof ThemedCardImage;
  Title: typeof ThemedCardTitle;
  FeaturedTitle: typeof ThemedCardFeaturedTitle;
  FeaturedSubTitle: typeof ThemedCardFeaturedSubtitle;
};

export const Card: CardProps = Object.assign(CardBase);

const ThemedCard = Object.assign(withTheme(Card, 'Card'), {
  Divider: ThemedCardDivider,
  Image: ThemedCardImage,
  Title: ThemedCardTitle,
  FeaturedTitle: ThemedCardFeaturedTitle,
  FeaturedSubtitle: ThemedCardFeaturedSubtitle,
});

export default ThemedCard;
