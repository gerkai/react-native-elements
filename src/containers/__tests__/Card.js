import React from 'react';
import {shallow} from 'enzyme';
import Card from '../Card';

describe('Card Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Card />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should have Card title', () => {
    const component = shallow(<Card title='Card Title' />);

    expect(component.props().children.props.children[0].props.children[0].props.children).toBe('Card Title');
  });
});
