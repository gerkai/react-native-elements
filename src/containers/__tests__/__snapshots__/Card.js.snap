// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e8ee",
              "borderWidth": 1,
              "margin": 15,
              "marginBottom": 0,
              "padding": 15,
              "shadowColor": "rgba(0,0,0, .2)",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            undefined,
            undefined,
          ]
    }
>
    <View
        style={
            Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                undefined,
              ]
        }
    >
        
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#e1e8ee",
                    "borderWidth": 1,
                    "margin": 15,
                    "marginBottom": 0,
                    "padding": 15,
                    "shadowColor": "rgba(0,0,0, .2)",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 1,
                  },
                  undefined,
                  undefined,
                ]
      }
>
      <View
            style={
                  Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                        undefined,
                      ]
            }
      >
            
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Card />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "white",
                                "borderColor": "#e1e8ee",
                                "borderWidth": 1,
                                "margin": 15,
                                "marginBottom": 0,
                                "padding": 15,
                                "shadowColor": "rgba(0,0,0, .2)",
                                "shadowOffset": Object {
                                  "height": 0,
                                  "width": 0,
                                },
                                "shadowOpacity": 1,
                                "shadowRadius": 1,
                              },
                              undefined,
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                        },
                                        undefined,
                                        undefined,
                                      ]
                    }
          >
                    
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "white",
                                "borderColor": "#e1e8ee",
                                "borderWidth": 1,
                                "margin": 15,
                                "marginBottom": 0,
                                "padding": 15,
                                "shadowColor": "rgba(0,0,0, .2)",
                                "shadowOffset": Object {
                                  "height": 0,
                                  "width": 0,
                                },
                                "shadowOpacity": 1,
                                "shadowRadius": 1,
                              },
                              undefined,
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Array [
                                        Object {
                                          "backgroundColor": "transparent",
                                        },
                                        undefined,
                                        undefined,
                                      ]
                    }
          >
                    
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Card />,
}
`;
